fix: allow empty type for Web Templates that are not sections

barredterra opened the pull request:
Currently, every **Web Template** needs to have a type of "Section", "Navbar" or "Footer".
However, a **Web Template** can also be a _component_. For example, a team member in a "Team" section or a navbar item. 
Wrapping some `<section>` HTML around these components results in a wrong html structure and layout.

This PR adds "Component" to the type options. **Web Templates** that are only a part of a "Section", "Navbar" or "Footer" should be of the "Component" type. There will be no additional HTML wrapped around a component. They are inserted into their parent **Web Template** as they are.

**Web Template** of type "Component"

- cannot be selected as a section of a **Web Page**.
- can be included in another **Web Template**, like this:

```
{{ web_block('My Component', {'heading': 'This is My Component', 'text': 'Lorem Ipsum ...'}) }}
```

barredterra committed:
fix: allow empty type for Web Templates that are not sections

netchampfaris reviewed (commented):
I like the idea of "Component" type. Also, if these web templates are allowed in Web Page, then the spacing modifier fields like Add Container, Add Space on Top etc should be hidden.

barredterra commented:
Good question. I didn't encounter any use case that requires components in a Web Page directly yet. Only as part of other Web Templates.

netchampfaris commented:
@barredterra Are you going to make the change of adding "Component" as a type?

barredterra commented:
Yes

barredterra committed:
Merge branch 'develop' into fix_web_template_type

barredterra committed:
fix: "Component" istead of empty type

barredterra committed:
feat: set filter on Web Page Blocks

Component is not allowed as a page section, only as part of another Web Template.

netchampfaris reviewed (changes requested):
I think we should be able to use Component blocks like this. 
```
{{ web_block('My Component', {'heading': 'This is My Component', 'text': 'Lorem Ipsum ...'}) }}
```

The default value for padding modifiers should be set according to the type of Web Template. In this case, they should be disabled for the type Component.

barredterra committed:
fix: disable padding and container by default

barredterra commented:
@netchampfaris I disabled add_padding / add_container in the backend, by default. In the frontend, in Web Page Blocks, they are still enabled by default. This produces the result we want:

- adding a web block via jinja adds no padding or container,
- adding a web block as section of a Web Page adds padding and container by default.

netchampfaris committed:
fix: Remove unnecessary code

- Extra spacing is only applied to templates of type Section
- should be able to call web_block without values
- default type should be Section
- set_query can be set once

netchampfaris commented:
@barredterra I made some changes.

Looks like we did not need to do much to have the API we wanted. Extra spacing is only applied to **Section** type templates.
"It just works"

Made passing values optional. Now we have a sweeter API.

```
{{ web_block('Component Name') }}
```

barredterra commented:
I see you enabled the padding / container again. Why? Will this be applied to components?

@netchampfaris

netchampfaris commented:
@barredterra 

No, it won't be applied:

https://github.com/frappe/frappe/blob/60d179eff7ecf5ca9ece3ab5c3c3554f69399fc1/frappe/templates/includes/web_block.html#L09-L21

netchampfaris committed:
Merge remote-tracking branch 'upstream/develop' into fix_web_template_type
